ENIGMAsystem/SHELL/Platforms/SDL/Window.cpp

```diff
@@ -276,18 +269,32 @@ void window_set_rectangle(int x, int y, int w, int h) {
 }
 
 int window_get_width() {
-  int w;
-  SDL_GetWindowSize(windowHandle, &w, nullptr);
-  return w;
+  //int w;
+  //SDL_GetWindowSize(windowHandle, &w, nullptr);
+  //return w;
+  int viewportWidth;
+  int viewportHeight;
+  SDL_GL_GetDrawableSize(windowHandle, &viewportWidth, &viewportHeight);
+  return viewportWidth;
+
 }
 
 int window_get_height() {
-  int h;
-  SDL_GetWindowSize(windowHandle, nullptr, &h);
-  return h;
+  //int h;
+ //SDL_GetWindowSize(windowHandle, nullptr, &h);
+ // return h;
+  int viewportWidth;
+  int viewportHeight;
+  SDL_GL_GetDrawableSize(windowHandle, &viewportWidth, &viewportHeight);
+  return viewportHeight;
+
 }
 
-void window_set_size(unsigned w, unsigned h) { SDL_SetWindowSize(windowHandle, w, h); }
+void window_set_size(unsigned w, unsigned h) { 
+ #ifndef _ANDROID_
```

Comment:
This is fine for now but before you're done you should remove that macro an abstract the function like you did for GLES